docs/ActiveLabel.rst

```diff
@@ -1,135 +1,320 @@
+For example, using the ``Actor`` ``ActiveLabel`` module:
+
+1. You can call ``person.actor?`` which will return true if the obj has the label associated with the ``Actor`` ``ActiveLabel``.
+2. You can call ``Person.actor.new`` or ``Person.actor.create`` to initialize / create a new ``Person`` instance with the additional ``Actor`` label.
+3. You can call ``Person.actor.all`` or ``Person.actor.first`` to return all ``Person`` nodes with the ``Actor`` label. In fact, calling ``Person.actor`` simply adds a label scope, which can be combined with any custom scopes you have (e.g. ``Person.most_popular`` -> ``Person.actor.most_popular``)
 
 Querying
 --------
 
-``ActiveLabel`` allows your Ruby module to act like a model class.  However, since you can add a label to any module, you can query for nodes across modules:
+Querying for ``ActiveLabel``s is easy, and can allow you to query across classes.
 
 .. code-block:: ruby
 
-  Destroyable.all
-
-  HasAddress.as(:obj).address.where(postal_code: '12345').pluck('DISTINCT obj')
+  # This returns all nodes which have the Actor label
+  Actor.all
 
-By default this returns all nodes for all models where the ``ActiveLabel`` module is defined.  If ``follows_label`` is declared, this returns just those nodes which have the label.
+  # This returns all nodes with the Director label which have a directed association to
+  # a node with the title "Star Wars"
+  # This works because the ``Director`` ``ActiveLabel`` defines a ``directed`` association
+  Director.as(:dir).directed.where(title: 'Star Wars').pluck('DISTINCT dir')
 
-By defining the ``follows_label``, some methods are automatically provided to allow you to filter and interrogate:
+Including an ``ActiveLabel`` module in a class will `automatically` add a few helper methods to the class and class instances.
 
 .. code-block:: ruby
 
-  Person.labeled_as_destroyed
+  Person.actor.all
+
+  Person.actor.first
 
-  Person.first.labeled_as_destroyed?
+Calling ``Person.actor`` simply adds a label scope, which can be combined with any custom scopes you have (e.g.
+``Person.most_popular`` -> ``Person.actor.most_popular``
 
 Associations
 ~~~~~~~~~~~~
 
-You can even create associations to traverse to labels:
+You can create associations with ActiveLabels:
 
 .. code-block:: ruby
 
-  class Organization
+  class Movie
     include Neo4j::ActiveNode
 
-    has_many :out, :addressables, type: :HAS_ADDRESSABLE_OBJECT, label_class: :HasAddress
+    has_many :in, :actors, type: :ACTS_IN, label_module: :Actor
 
-    # `model_class` acts as a filter to the `label_class` argument.  Both `model_class` and `label_class` can be arrays
-    has_many :out, :addressable_people, type: :HAS_ADDRESSABLE_OBJECT, label_class: :HasAddress, model_class: :Person
+    # `label_module` acts as a filter to the `model_class` argument.
+    # Both `model_class` and `label_module` can be arrays
+    has_many :in, :human_actors, type: :ACTS_IN, label_module: :Actor, model_class: :Person
```

Comment:
Where you had *"model_class" acts as a filter...* I changed it to *"label_module" acts as a filter* because the class is what is instantiated and so seems like the more "primary" descriptor (in my mind). Put another way, `ActiveLabel` is marking up `ActiveNode`. Also I changed from *"label_class"* to *"label_module"*, because the labels are modules rather than classes.